src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -72,71 +97,76 @@ func (sm *StorageMinerActor_I) submitFaultReport(
 		NewTerminatedFaults_: newTerminatedFaults,
 	}
 
-	sm.SectorTable().Impl().TerminationFaultCount_ = util.UVarint(0)
-
 	panic(faultReport)
 	// TODO: Send(SPA, ProcessFaultReport(faultReport))
+
+	h, st := a.State(rt)
+	st.SectorTable().Impl().TerminationFaultCount_ = util.UVarint(0)
+	h.Commit(st)
 }
 
 // construct PowerReport from SectorTable
-func (sm *StorageMinerActor_I) submitPowerReport() {
+func (a *StorageMinerActorCode_I) _submitPowerReport(rt Runtime) {
+	h, st := a.State(rt)
 	powerReport := &power.PowerReport_I{
-		ActivePower_:   sm.SectorTable().ActivePower(),
-		InactivePower_: sm.SectorTable().InactivePower(),
+		ActivePower_:   st.SectorTable().ActivePower(),
+		InactivePower_: st.SectorTable().InactivePower(),
 	}
+	h.Release(st)
 
 	panic(powerReport)
 	// TODO: Send(SPA, ProcessPowerReport(powerReport))
 }
 
-func (sm *StorageMinerActor_I) onMissedPoSt() {
-
-	failingSectorNumbers := getSectorNums(sm.Sectors())
+func (a *StorageMinerActorCode_I) _onMissedPoSt(rt Runtime) {
+	h, st := a.State(rt)
+	failingSectorNumbers := getSectorNums(st.Sectors())
 	for _, sectorNo := range failingSectorNumbers {
-		sm.failSector(sectorNo, true)
+		st._updateFailSector(rt, sectorNo, true)
 	}
+	st._updateExpireSectors(rt)
+	h.Commit(st)
 
-	sm.expireSectors()
+	h, st = a.State(rt)
+	newDetectedFaults := st.SectorTable().FailingSectors()
+	newTerminatedFaults := st.SectorTable().TerminationFaultCount()
+	h.Release(st)
```

Comment:
nice -- is the idea that calling `st.SectorTable()` after `h.Release()` would panic? (i'd like that...)

Reply:
Yes (or more precisely, would have the same effect as `rt.Fatal(.)`)